src/soundpython/audio.py

```diff
@@ -321,6 +321,75 @@ def get_channel(self, channel: int) -> Audio:
 
         return Audio(channel_data, new_metadata)
 
+    def resample(self, target_sample_rate: int) -> Audio:
+        """
+        Resample the audio to a new sample rate
+
+        Args:
+            target_sample_rate: New sample rate in Hz
+
+        Returns:
+            Audio: New Audio instance with resampled audio
+        """
+        if target_sample_rate == self.metadata.sample_rate:
+            return self
+
+        # Calculate resampling ratio
+        ratio = target_sample_rate / self.metadata.sample_rate
+
+        target_length = round(self.data.shape[0] * ratio)
+
+        audio_array = self.data
+        if self.metadata.channels == 1:
+            audio_array = audio_array.reshape(-1, 1)
+
+        resampled_data = np.zeros((target_length, self.metadata.channels), dtype=np.float32)
+
+        for channel in range(self.metadata.channels):
+            resampled_data[:, channel] = self._resample_channel(audio_array[:, channel], target_length)
+
+        new_metadata = AudioMetadata(
+            sample_rate=target_sample_rate,
+            channels=self.metadata.channels,
+            sample_width=self.metadata.sample_width,
+            duration_seconds=target_length / target_sample_rate,
+            frame_count=target_length,
+        )
+        if self.metadata.channels == 1:
+            resampled_data = resampled_data.flatten()
+
+        return Audio(resampled_data, new_metadata)
+
+    def _resample_channel(self, data: np.ndarray, new_length: int, axis: int = 0) -> np.ndarray:
```

Comment:
Looking at our code, we always pass 1D array to this method, because we subslice single channel in line 349: `self._resample_channel(audio_array[:, channel], target_length)`

Would it make sense to call this e.g. `_resample_1d_array` and assume that `axis` is always 0, without making it an argument?

Reply:
Done, I think the name`_resample_channel` still makes sense (probably even more now), as we resample just one channel